core/state_prefetcher.go

```diff
@@ -50,43 +48,45 @@ func NewStatePrefetcher(config *params.ChainConfig, bc *BlockChain, engine conse
 // Prefetch processes the state changes according to the Ethereum rules by running
 // the transaction messages using the statedb, but any changes are discarded. The
 // only goal is to pre-cache transaction signatures and snapshot clean state.
-func (p *statePrefetcher) Prefetch(block *types.Block, statedb *state.StateDB, cfg vm.Config, interrupt *uint32) {
+func (p *statePrefetcher) Prefetch(block *types.Block, statedb *state.StateDB, cfg *vm.Config, interruptCh <-chan struct{}) {
 	var (
 		header = block.Header()
 		signer = types.MakeSigner(p.config, header.Number)
 	)
 	transactions := block.Transactions()
-	sortTransactions := make([][]*types.Transaction, prefetchThread)
-	for i := 0; i < prefetchThread; i++ {
-		sortTransactions[i] = make([]*types.Transaction, 0, len(transactions)/prefetchThread)
-	}
-	for idx := range transactions {
-		threadIdx := idx % prefetchThread
-		sortTransactions[threadIdx] = append(sortTransactions[threadIdx], transactions[idx])
-	}
+	txChan := make(chan int, prefetchThread)
 	// No need to execute the first batch, since the main processor will do it.
 	for i := 0; i < prefetchThread; i++ {
-		go func(idx int) {
+		go func() {
 			newStatedb := statedb.Copy()
 			newStatedb.EnableWriteOnSharedStorage()
 			gaspool := new(GasPool).AddGas(block.GasLimit())
 			blockContext := NewEVMBlockContext(header, p.bc, nil)
-			evm := vm.NewEVM(blockContext, vm.TxContext{}, statedb, p.config, cfg)
+			evm := vm.NewEVM(blockContext, vm.TxContext{}, statedb, p.config, *cfg)
```

Comment:
`cfg` is not a safe way to use here.
NewEVM-->NewEVMInterpreter-->`for i, eip := range cfg.ExtraEips`
this part of code would modify a slice variable, which could be modified concurrently by `bc.processor.Process`, this unsafe use also occurred in pr #803
what's more: the logic in `for i, eip := range cfg.ExtraEips` is not correct, adjacent errors would be skipped

Reply:
accept, will be merged after https://github.com/bnb-chain/bsc/pull/953/